mlrun/feature_store/retrieval/job.py

```diff
@@ -37,17 +44,20 @@ def run_merge_job(
         raise mlrun.errors.MLRunInvalidArgumentError("target object must be specified")
     name = f"{name}_merger"
     run_config = run_config or RunConfig()
+    kind = run_config.kind or ("spark" if engine == "spark" else "job")
     if not run_config.function:
         function_ref = vector.spec.function.copy()
         if function_ref.is_empty():
-            function_ref = FunctionReference(name=name, kind="job")
+            function_ref = FunctionReference(name=name, kind=kind)
         if not function_ref.url:
-            function_ref.code = _default_merger_handler
+            function_ref.code = _default_merger_handler.replace(
+                "{{{engine}}}", merger.__name__
+            ).replace("{{{engine_args}}}", str(engine_args or {}))
         run_config.function = function_ref
 
-    function = run_config.to_function(
-        "job", mlrun.mlconf.feature_store.default_job_image
-    )
+    function = run_config.to_function(kind, merger.get_default_image(kind))
+    if run_config.kind == RuntimeKinds.remotespark:
+        function.with_spark_service(spark_service=spark_service)
```

Comment:
What happens if `spark_service` is `None`? Don't we need to explode in this case?